server/src/main/java/org/opensearch/search/startree/StarTreeQueryContext.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.startree;
+
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.util.FixedBitSet;
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.index.codec.composite.CompositeIndexFieldInfo;
+
+import java.util.Map;
+
+/**
+ * Query class for querying star tree data structure.
+ *
+ * @opensearch.experimental
+ */
+@ExperimentalApi
+public class StarTreeQueryContext {
+
+    /**
+     * Star tree field info
+     * This is used to get the star tree data structure
+     */
+    private final CompositeIndexFieldInfo starTree;
+
+    /**
+     * Map of field name to a value to be queried for that field
+     * This is used to filter the data based on the query
+     */
+    private final Map<String, Long> queryMap;
+
+    /**
+    * Cache for leaf results
+    * This is used to cache the results for each leaf reader context
+    * to avoid reading the filtered values from the leaf reader context multiple times
+    */
+    private FixedBitSet[] starTreeValues;
+
+    public StarTreeQueryContext(CompositeIndexFieldInfo starTree, Map<String, Long> queryMap, int cacheStarTreeValuesSize) {
```

Comment:
The name of the parameter should be more specific than `cacheStarTreeValuesSize`. Maybe `numLeafReaderContexts`. Of course, then using a value `<= 0` as a way of disabling the cache may be confusing.

The important thing is that it is exactly the number of `LeafReaderContexts` in the `IndexReader`. Otherwise the the `get`/`set` methods won't be guaranteed to work.

Reply:
Didn't want to increase the number of parameters in the constructor just to pass a boolean, since just calling it `numLeafReaderContexts` will then be confusing to use the logic with `>=0`.

How about calling it `numLeafReaderContextCache` where `-1` can continue to signify we don't need cache.